staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -524,9 +525,10 @@ func (s *store) List(ctx context.Context, key, resourceVersion string, pred stor
 		options = append(options, clientv3.WithRange(rangeEnd))
 		key = continueKey
 
-		options = append(options, clientv3.WithRev(continueRV))
-		returnedRV = continueRV
-
+		if continueRV > 0 {
```

Comment:
This method is already complex, and this makes it harder to read.  Add a comment.